packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1127,8 +1127,43 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// ```
   ///
   /// It is an error to call this method after the framework calls [dispose].
-  /// You can determine whether it is legal to call this method by checking
+  /// You can determine whether it is legal to call [setState] by checking
   /// whether the [mounted] property is true.
+  ///
+  /// # Performance considerations
+  ///
+  /// Calls to [setState] cause a rebuild of the current widget and,
+  /// potentially, all of its children. This isn't efficient and generally not
```

Comment:
+1, see also my previous comment. We should not scare people from using setState.

Reply:
My intention is not to scare people and give the impression that setState is a bad beast.

I just see that the Flutter inline documentation doesn't give much inputs about rebuilds and builder functions. I am very happy that we merged [#9862](https://github.com/flutter/flutter/pull/98628), which adds more context around `AnimatedBuilder` and `Listenable`.

Now I would also add these lines to point out that there are **other** options to setState. One might not be aware of value notifiers or the animatedbuilder + listenable thing and so were we point it out🙂 

I think that it can be helpful and it also guides the user to read more documentation and gives advices on where to look for more info about state management (in the docstrings)